### ISSUE TITLE
Add CI and releases

### ISSUE THREAD

**Issue by m8mble** [nghttp2/nghttp2-asio#12]
Currently, this library neither offers tests, CI or releases (as versioned milestones). This would be beneficial for a variety of reasons 

 - future adaptations to the functionality (without breaking anything)
 - making the library easier to approach for developers
 - integrating its functionality (e.g. into package managers)

I'd volunteer to add CI once there are some tests (if this would be desirable).

**Comment by patlecat** [comment]
I totally agree. But for some reason no updates have been done here for almost a year and even PullRequests are ignored.
